Mathlib/Algebra/Order/Nonneg/GeomSum.lean

```diff
@@ -0,0 +1,28 @@
+/-
+Copyright (c) 2024 Johan Commelin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johan Commelin
+-/
+
+import Mathlib.Algebra.Order.Nonneg.Field
+import Mathlib.Algebra.GeomSum
+import Mathlib.Data.Fintype.BigOperators
+
+variable {α : Type*} [LinearOrderedField α]
+
+lemma Nonneg.geom_sum_of_one_lt {a : {x : α // 0 ≤ x}} (h : 1 < a) (n : ℕ) :
+    ∑ i ∈ Finset.range n, a ^ i = (a ^ n - 1) / (a - 1) := by
```

Comment:
Is it true in a `LinearOrderedSemifield` + `OrderedSub`? Also, could you please add the `tsub_mul_geom` and `geom_2` versions?

Reply:
I'm not sure about a `LinearOrderedSemifield`, but I generalized it to `CanonicallyLinearOrderedSemifield`

Reply:
I'm not sure if it should remain in `Nonneg` now, though